_inc/lib/core-api/wpcom-endpoints/publicize-connection-test-results.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+require_once dirname( __FILE__ ) . '/publicize-connections.php';
+
+/**
+ * Publicize: List Connection Test Result Data
+ *
+ * All the same data as the Publicize Connections Endpoint, plus test results.
+ *
+ * @since 6.8
+ */
+class WPCOM_REST_API_V2_Endpoint_List_Publicize_Connection_Test_Results extends WPCOM_REST_API_V2_Endpoint_List_Publicize_Connections {
+	public function __construct() {
+		$this->namespace = 'wpcom/v2';
+		$this->rest_base = 'publicize/connection-test-results';
+
+		add_action( 'rest_api_init', array( $this, 'register_routes' ) );
+	}
+
+	/**
+	 * Called automatically on `rest_api_init()`.
+	 */
+	public function register_routes() {
+		register_rest_route( $this->namespace, '/' . $this->rest_base, array(
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_items' ),
+				'permission_callback' => array( $this, 'get_items_permission_check' ),
+			),
+			'schema' => array( $this, 'get_public_item_schema' ),
+		) );
+	}
+
+	/**
+	 * Adds the test results properties to the Connection schema.
+	 */
+	public function get_item_schema() {
+		$schema = array(
+			'$schema' => 'http://json-schema.org/draft-04/schema#',
+			'title' => 'jetpack-publicize-connection-test-results',
+			'type' => 'object',
+			'properties' => $this->get_connection_schema_properties() + array(
+				'test_success' => array(
+					'description' => __( 'Did the Publicize Connection test pass?', 'jetpack' ),
+					'type' => 'boolean'
+				),
+				'test_message' => array(
+					'description' => __( 'Publicize Connection success or error message', 'jetpack' ),
+					'type' => 'string',
+				),
+				'can_refresh' => array(
+					'description' => __( 'Can the current user refresh the Publicize Connection?', 'jetpack' ),
+					'type' => 'boolean',
+				),
+				'refresh_text' => array(
+					'description' => __( 'Message instructing the user to refresh their Connection to the Publicize Service', 'jetpack' ),
+					'type' => 'string',
+				),
+				'refresh_url' => array(
+					'description' => __( 'URL for refreshing the Connection to the Publicize Service', 'jetpack' ),
+					'type' => 'string',
+					'format' => 'uri',
+				),
+			),
+		);
+
+		return $this->add_additional_fields_schema( $schema );
+	}
+
+	/**
+	 * @param WP_REST_Request
+	 * @see Publicize::get_publicize_conns_test_results()
+	 * @return WP_REST_Response suitable for 1-page collection
+	 */
+	public function get_items( $request ) {
+		global $publicize;
```

Comment:
How about setting a `$publicize` instance var upon construction? We'll probably still use the global for that, but at least we'd move responsibility to the consumer, arguably providing _somewhat_ better encapsulation, and possibly facilitating cleaner separation of concerns in a hypothetical future without that global...

Reply:
The `$publicize` global isn't set by the time this class is constructed (at least, the way these files are currently loaded). Getting things loaded in the right order would take a fair bit of work. Let's wait for that hypothetical refactor in the future ;)